### DISCUSSION TITLE
https://vuepress-theme-hope.github.io/v2/zh/guide/advanced/sitemap.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#925] [Announcements]
# https://vuepress-theme-hope.github.io/v2/zh/guide/advanced/sitemap.html

vuepress-theme-hope 通过内置 @vuepress/plugin-sitemap 为你提供 Sitemap 生成。 插件会为你的网站自动生成 Sitemap，它会自动根据页面的 Git 的时间戳生成页面的最后更新时间，同时会根据站点的多语言配置声明页面的其他语言替代版本地址。 如果你不需要这个插件，请在主题选项中设置 plugins....

https://vuepress-theme-hope.github.io/v2/zh/guide/advanced/sitemap.html

**Comment by Seven-97** [comment]
这个是如何获取到sitemap.xml文件的？我设置了hostname之后，访问站点下的 sitemap.xml ，会自动渲染成sitemap.xml.html，然后就404了

**Reply by Mister-Hope** [reply]
你可能需要一个正确的nginx配置

**Comment by yuanmoc** [comment]
> 如果你不希望某些页面出现在 sitemap 中，你可以将它们变成一个数组传入到 excludeUrls，或者通过 filter 传入一个过滤函数。你也可以在页面的 frontmatter 中设置 sitemap.exclude。

使用 filter 配置有效么？我配置后console.error信息都没有打印出来。配置代码如下：

```js
  plugins: {
    sitemap: {
      hostname: 'https://yourdomain.com',
      filter: (page) => {
        console.error("执行【sitemap】filter")
        // 排除特定路径的页面
        return page.path.includes('/posts/');
      }
    },
}
```
